presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -602,6 +604,27 @@ public void createTable(Session session, String catalogName, ConnectorTableMetad
         metadata.createTable(session.toConnectorSession(connectorId), tableMetadata, ignoreExisting);
     }
 
+    @Override
+    public TableHandle createTemporaryTable(Session session, String catalogName, List<ColumnMetadata> columns, Optional<PartitioningMetadata> partitioningMetadata)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogName);
+        ConnectorId connectorId = catalogMetadata.getConnectorId();
+        ConnectorMetadata metadata = catalogMetadata.getMetadata();
+        ConnectorTableHandle connectorTableHandle = metadata.createTemporaryTable(
+                session.toConnectorSession(connectorId),
+                columns,
+                partitioningMetadata.map(partitioning -> createConnectorPartitioningMetadata(connectorId, partitioning)));
+        return new TableHandle(connectorId, connectorTableHandle);
+    }
+
+    private static ConnectorPartitioningMetadata createConnectorPartitioningMetadata(ConnectorId connectorId, PartitioningMetadata partitioningMetadata)
+    {
+        ConnectorId partitioningConnectorId = partitioningMetadata.getPartitioningHandle().getConnectorId()
+                .orElseThrow(() -> new VerifyException("connectorId is expected to be present in the connector partitioning handle"));
```

Comment:
I'm not sure if this should be `IllegalStateException`.

Reply:
IllegalStateException is intended to indicate that the method cannot be called due to an illegal object state. let me actually change it to `IllegalArgumentException`